src/main/java/frc/robot/motors/FeedMotorControllerEncoderGroup.java

```diff
@@ -0,0 +1,165 @@
+
+    /**
+     * Create a new MotorControllerGroup with the provided MotorControllers and feed
+     * constants.
+     *
+     * @param motorController  The first MotorControllerEncoder to add
+     * @param motorControllers The MotorControllerEncoder to add
+     */
+    public FeedMotorControllerEncoderGroup(
+            MotorControllerEncoder motorController, MotorControllerEncoder... motorControllers) {
+        m_motorControllers = new MotorControllerEncoder[motorControllers.length + 1];
+        m_motorControllers[0] = motorController;
+        System.arraycopy(motorControllers, 0, m_motorControllers, 1, motorControllers.length);
+        init();
+    }
+
+    public FeedMotorControllerEncoderGroup(MotorControllerEncoder[] motorControllers) {
+        m_motorControllers = Arrays.copyOf(motorControllers, motorControllers.length);
+        init();
+    }
+
+    public void initFF(FeedforwardConstant forward, FeedforwardConstant backward, double maxSpeed, double maxAcceleration) {
+        this.forwardFF = new SimpleMotorFeedforward(forward.getkS(), forward.getkV(), forward.getkA());
+        this.backwardFF = new SimpleMotorFeedforward(backward.getkS(), backward.getkV(), backward.getkA());
+        this.maxSpeed = maxSpeed;
+        this.maxAcceleration = maxAcceleration;
+    }
+
+    public void initFB(FeedbackConstant forwardVelocity, FeedbackConstant backwardVelocity,
+            FeedbackConstant forwardPosition,
+            FeedbackConstant backwardPosition) {
+        this.forwardVelocityFB = new PIDController(forwardVelocity.getkP(), 0.0, forwardVelocity.getkD());
+        this.backwardVelocityFB = new PIDController(backwardVelocity.getkP(), 0.0, backwardVelocity.getkD());
+        this.forwardPositionFB = new PIDController(forwardPosition.getkP(), 0.0, forwardPosition.getkD());
+        this.backwardPositionFB = new PIDController(backwardPosition.getkP(), 0.0, backwardPosition.getkD());
+    }
+
+    private void init() {
+        for (MotorControllerEncoder controller : m_motorControllers) {
+            SendableRegistry.addChild(this, controller);
+        }
+        instances++;
+        SendableRegistry.addLW(this, "FeedMotorControllerGroup", instances);
+    }
+
+    @Override
+    public void close() {
+        SendableRegistry.remove(this);
+    }
+
+    public void set(double speed, boolean velocity) {
+        double setpoint = (m_isInverted ? -speed : speed) * maxSpeed;
+        double currentVelocity = m_motorControllers[0].getVelocity();
+        boolean forward = setpoint >= currentVelocity;
+        double ffVoltage = 0;
+
+        if (Math.abs(setpoint - currentVelocity) >= maxAcceleration * 0.02) {
+            setpoint = forward ? currentVelocity + (maxAcceleration * 0.02) : currentVelocity  - (maxAcceleration * 0.02);
+        }
+
+        ffVoltage = forward ? forwardFF.calculate(setpoint) : backwardFF.calculate(setpoint);
+
+        PIDController selectedController = null;
+        if (velocity) {
```

Comment:
This would read better if the variable was called "useVelocityControl" or something similar.